pkg/image/prune/prune.go

```diff
@@ -302,13 +313,15 @@ func addImagesToGraph(g graph.Graph, images *imageapi.ImageList, algorithm prune
 
 		glog.V(4).Infof("Examining image %q", image.Name)
 
-		if image.Annotations == nil {
-			glog.V(4).Infof("Image %q with DockerImageReference %q belongs to an external registry - skipping", image.Name, image.DockerImageReference)
-			continue
-		}
-		if value, ok := image.Annotations[imageapi.ManagedByOpenShiftAnnotation]; !ok || value != "true" {
-			glog.V(4).Infof("Image %q with DockerImageReference %q belongs to an external registry - skipping", image.Name, image.DockerImageReference)
-			continue
+		if !algorithm.allImages {
+			if image.Annotations == nil {
+				glog.V(4).Infof("Image %q with DockerImageReference %q belongs to an external registry - skipping", image.Name, image.DockerImageReference)
+				continue
+			}
+			if value, ok := image.Annotations[imageapi.ManagedByOpenShiftAnnotation]; !ok || value != "true" {
```

Comment:
The two ifs can be squashed to just one `if !strings.Contains(image.Annotations[imageapi.ManagedByOpenShiftAnnotation], "true") { ...`.